rel8-1.7.0.0 fails to build with semialign >= 1.3: missing Unzip (First a) instance

## Problem

`rel8-1.7.0.0` fails to compile when `semialign >= 1.3` is resolved. `semialign-1.3` made `Unzip` a superclass of `Semialign`, but `Rel8.Schema.HTable.Vectorize` defines `instance Semialign (First a)` without a corresponding `Unzip (First a)` instance.

**Error:**
```
src/Rel8/Schema/HTable/Vectorize.hs:200:10: error: [GHC-39999]
    • No instance for 'Unzip (First a)'
        arising from the superclasses of an instance declaration
    • In the instance declaration for 'Semialign (First a)'
    |
200 | instance Semialign (First a) where
    |          ^^^^^^^^^^^^^^^^^^^
```

## Root cause

In `semialign >= 1.3`, the class hierarchy changed so that `Semialign` requires `Unzip` as a superclass (`class Unzip f => Semialign f`). The `Unzip` import in `Vectorize.hs` also omits `(..)`, so the methods are not in scope.

## Who is affected

Any project that depends on both `rel8` and a package that transitively requires `semialign >= 1.3`. In practice, `aeson >= 2.2` pulls in `semialign >= 1.3`, so users of `rel8` + `aeson >= 2.2` on a clean build will hit this.

rel8's own `cabal.project` uses `allow-newer: *:base` which forces `semialign-1.2.x` and hides the issue.

## Fix

Two changes to `src/Rel8/Schema/HTable/Vectorize.hs`:

1. Import `Unzip` with its methods: `import Data.Zip (Unzip (..), Zip, Zippy(..), zipWith)`
2. Add the missing instance before `Semialign (First a)`:

```haskell
instance Unzip (First a) where
  unzipWith _ (First a) = (First a, First a)
```

A working patch is available at: https://github.com/MMZK1526/rel8/tree/fix-unzip (based on the `0e1acf0` / v1.7.0 commit).

## Environment

- GHC 9.6.7
- rel8 1.7.0.0
- semialign 1.3+ (e.g. 1.4)
- aeson 2.2+

--- a/changelog.d/20260616_150020_cgeorgii_support_ghc_9_14.md
+++ b/changelog.d/20260616_150020_cgeorgii_support_ghc_9_14.md
@@ -1,0 +1,3 @@
+### Added
+
+- Support GHC-9.14 and `semialign >= 1.4`.

--- a/rel8.cabal
+++ b/rel8.cabal
@@ -21,7 +21,7 @@ library
   build-depends:
       aeson
     , attoparsec
-    , base >= 4.16 && < 4.22
+    , base >= 4.16 && < 4.23
     , base16 >= 1.0
     , base-compat >= 0.11 && < 0.15
     , bifunctors

--- a/src/Rel8/Schema/HTable/Vectorize.hs
+++ b/src/Rel8/Schema/HTable/Vectorize.hs
@@ -67,7 +67,7 @@ import Rel8.Type.Information ( TypeInformation )
 
 -- semialign
 import Data.Align (Semialign, alignWith)
-import Data.Zip (Unzip, Zip, Zippy(..), zipWith)
+import Data.Zip (Unzip, Zip, Zippy(..), unzipWith, zipWith)
 
 -- semigroupoids
 import Data.Functor.Apply (Apply)
@@ -203,3 +203,7 @@ instance Semialign (First a) where
 
 instance Zip (First a) where
   zipWith _ (First a) _ = First a
+
+
+instance Unzip (First a) where
+  unzipWith _ (First a) = (First a, First a)

--- a/src/Rel8/Statement.hs
+++ b/src/Rel8/Statement.hs
@@ -8,6 +8,7 @@
 {-# language RecordWildCards #-}
 {-# language ScopedTypeVariables #-}
 {-# language StandaloneKindSignatures #-}
+{-# language TypeAbstractions #-}
 {-# language TypeApplications #-}
 
 module Rel8.Statement

